examples/data_types_and_io/data_types_and_io/custom_objects.py

```diff
@@ -10,12 +10,12 @@
 # Flyte supports passing JSON between tasks. But to simplify the usage for the users and introduce type-safety,
 # Flytekit supports passing custom data objects between tasks.
 #
-# Currently, data classes decorated with `@dataclass_json` are supported.
+# Currently, data classes inherited from DataClassJSONMixin are supported.
 # One good use case of a data class would be when you want to wrap all input in a data class in the case of a map task
 # which can only accept one input and produce one output.
 #
 # This example shows how users can serialize custom JSON-compatible dataclasses between successive tasks using the
-# excellent [dataclasses_json](https://pypi.org/project/dataclasses-json/) library.
+# excellent [mashumaro](https://github.com/Fatal1ty/mashumaro) library.
```

Comment:
And add a `:::{note}` to tell people use `mashumaro.DataClassJSONMixin` if they are using`flytekit 1.10`, otherwise use `dataclasses_json`